### ISSUE TITLE
Performance impact when using proxyfmu

### ISSUE THREAD

**Issue by davidhjp01** [open-simulation-platform/libcosim#722]
Performance difference between simulations with and without proxyfmu is quite significant when tested with the tutorial example (spring-mass-damper) using `cosim-cli`:

Without proxyfmu (endtime 360, stepsize 0.01):
![image](https://user-images.githubusercontent.com/88651540/200189614-c2e5643e-9076-4357-9337-f7d1ab68fea1.png)

With proxyfmu (endtime 360, stepsize 0.01, ran on localhost):
![image](https://user-images.githubusercontent.com/88651540/200189628-b91744b3-8c41-4c2a-9756-f1ac56fa36f7.png)

The running time when using proxyfmu includes fmu initialization time (10 seconds? max) but it is still 15 times slower.

Given fmus need to exchange data in every step (including `doStep` and other house keeping communications), such performance degradation is expected. But need some discussion if there is something that can be optimized (or replace thrift?) to improve the simulation speed.

**Comment by kyllingstad** [comment]
> A low hanging fruit here is to reduce the amount of get/set calls by collecting all the types in a single get/set call rather than one for real/integer/boolean/string.

I agree. This is [how `async_slave` was designed](https://github.com/open-simulation-platform/libcosim/blob/194a5daa780e9b0d9520cb1e54e4c8caf6cf85a0/include/cosim/async_slave.hpp#L238). The idea was that `async_slave` would enable efficient communication with remote slaves, while `slave` was a lower-level interface for local slaves—basically a layer on top of the FMI C interface.

Now that `slave` is our only slave API, it may be time to improve it.
